### ISSUE TITLE
Mac M1 Mini failure to launch

### ISSUE THREAD

**Issue by cieltian** [realAbitbol/BardMac#5]
Hello! I have a M1 silicon Mac mini, every time I open the bardmac, I get a failure to launch with the debugging menu for py2app, I have tried launching directly from the terminal as well, I am having no luck. I have tried downloading it multiple ways, multiple times, and am getting frustrated. I was hoping maybe I am missing something easy. 

![Screen Shot 2021-12-08 at 12 02 35 AM](https://user-images.githubusercontent.com/95734649/145151211-5895d63d-0a29-4766-b50f-4e4c6d456a15.png)

**Comment by cbackas42** [comment]
I had this problem too, was able to extract the stack trace which mainly seems to be that TCL is not available. I'm not clear if that's supposed to be part of the packaging or if it's a prerequisite to using this to install it yourself...

Traceback (most recent call last):
  File "/Applications/BardMac.app/Contents/Resources/__boot__.py", line 130, in <module>
    _run()
  File "/Applications/BardMac.app/Contents/Resources/__boot__.py", line 84, in _run
    exec(compile(source, path, "exec"), globals(), globals())
  File "/Applications/BardMac.app/Contents/Resources/BardMac.py", line 23, in <module>
    import PySimpleGUI as Sg
  File "/Applications/BardMac.app/Contents/Resources/lib/python3.9/PySimpleGUI/__init__.py", line 2, in <module>
    from .PySimpleGUI import *
  File "/Applications/BardMac.app/Contents/Resources/lib/python3.9/PySimpleGUI/PySimpleGUI.py", line 125, in <module>
    tclversion_detailed = tkinter.Tcl().eval('info patchlevel')
  File "tkinter/__init__.pyc", line 2372, in Tcl
  File "tkinter/__init__.pyc", line 2270, in __init__
_tkinter.TclError: Can't find a usable init.tcl in the following directories: 
    /usr/local/Cellar/tcl-tk/8.6.12/lib/tcl8.6 /Applications/BardMac.app/Contents/lib/tcl8.6 /Applications/BardMac.app/lib/tcl8.6 /Applications/BardMac.app/Contents/library /Applications/BardMac.app/library /Applications/BardMac.app/tcl8.6.12/library /Applications/tcl8.6.12/library



This probably means that Tcl wasn't installed properly.

2021-12-08 18:47:33.674 BardMac[51075:10645409] Launch error
2021-12-08 18:47:33.674 BardMac[51075:10645409] Launch error
See the py2app website for debugging launch issues

**Comment by cieltian** [comment]
Hey thank you for the reply! At least I am not alone here. I am guessing this is the same issue that I am having. I am going to look into this further while I wait. Were you able to get yours resolved or are you still having issues?

**Comment by cbackas42** [comment]
No, I have yet to get it to work at all. I tried launching it directly with Python and get:
cbackas@Phoenix ~/D/BardMac-1.0.6> python3 BardMac.py 
Traceback (most recent call last):
  File "BardMac.py", line 21, in <module>
    from typing import Final
ImportError: cannot import name 'Final' from 'typing' (/Applications/Xcode.app/Contents/Developer/Library/Frameworks/Python3.framework/Versions/3.7/lib/python3.7/typing.py)

I also tried building the py2app thing myself, but it runs into the same problem as seen here. I did install tkinter and all the other listed pip modules too.

**Comment by realAbitbol** [comment]
Hey, I think your python version is a bit too old, please try with a newer version so you'll get the necessary import for the Final thing.
You can download python here : https://www.python.org/downloads/

**Comment by realAbitbol** [comment]
Regarding the m1 problem, I'm looking into it. It maybe is an incompatibility problem with py2app but I'm not sure yet and I don't have a M1 mac (yet).

**Comment by realAbitbol** [comment]
Tried adding an option to force running on rosetta on m1 macs.
Can you try the attached app 
[BardMac.app.zip](https://github.com/realAbitbol/BardMac/files/7687301/BardMac.app.zip)
?

**Comment by cieltian** [comment]
I tried running that just now, it still doesn't work. Still getting the launch error. I also am getting the "This probably means that Tcl wasn't installed properly." error when I trace it back to the launch error. "Can't find a usable init.tcl in the following directories: "

**Comment by realAbitbol** [comment]
https://py2app.readthedocs.io/en/latest/faq.html?highlight=arch#frequently-asked-questions

I think that's the way. I'll try tonight to download and package the x86_64 versions of the libraries too within the app.

**Comment by cieltian** [comment]
Thank you so much by the way for attempting to figure this out!

**Comment by cieltian** [comment]
So, I tried that zip you posted in the other problem thread where you tried forcing it to run on rosetta, I saw you hadn't gotten an answer yet but I'm getting the same result. Didn't want to take over their thread but figured that might be helpful if you were waiting for an answer over there. I'm not sure if you posted that over there first or posted this here first.

**Comment by cieltian** [comment]
Any update on this yet that I might have missed?
